confmap/internal/compose_hook.go

```diff
@@ -0,0 +1,102 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
```

Comment:
I feel like I should know this with more certainty, but shouldn't we include an attribution pointing to the [original license](https://github.com/go-viper/mapstructure/blob/c29fc28e7927f11614d6f7e0fbf108a5c75750bb/LICENSE) here somehow?

Reply:
Ah, right, I modified the license and copyright header and stored it in a separate folder